linux/deploy_vm/rebuild_ubuntu_unattend_install_iso.yml

```diff
@@ -0,0 +1,178 @@
+        files:
+          - "isolinux/isolinux.cfg"
+      ignore_errors: true
+      register: result_extract_file
+
+    - name: "Modify isolinux.cfg for some old releases such as Ubuntu 20.04 when the firmware is bios"
+      block:
+        - name: "Modify boot option"
+          ansible.builtin.blockinfile:
+            path: "{{ src_iso_file_dir }}/isolinux.cfg"
+            insertafter: "^#.*etc.*"
+            block: |
+              default live-install
+              label live-install
+                menu label ^Install Ubuntu
+                kernel /casper/vmlinuz
+                append  file=/cdrom/preseed/ubuntu.seed boot=casper debug-ubiquity automatic-ubiquity initrd=/casper/initrd quiet splash noprompt --- console=ttyS0,115200n8
+
+        - name: "Print the content of modified file isolinux.cfg"
+          ansible.builtin.debug:
+            msg: "{{ lookup('file', src_iso_file_dir + '/isolinux.cfg') }}"
+
+        - name: "Customize the ISO with isolinux/isolinux.cfg"
+          community.general.iso_customize:
+            src_iso: "{{ src_iso_file_path }}"
+            dest_iso: "{{ rebuilt_unattend_iso_path }}"
+            add_files:
+              - src_file: "{{ src_iso_file_dir }}/grub.cfg"
+                dest_file: "/boot/grub/grub.cfg"
+              - src_file: "{{ new_unattend_install_conf }}"
+                dest_file: "/preseed/ubuntu.seed"
+              - src_file: "{{ src_iso_file_dir }}/isolinux.cfg"
+                dest_file: "/isolinux/isolinux.cfg"
+      when: 
+        - result_extract_file is defined
+        - result_extract_file.failed is defined
+        - not result_extract_file.failed
+
+    - name: "Customize the ISO without isolinux/isolinux.cfg"
+      community.general.iso_customize:
+        src_iso: "{{ src_iso_file_path }}"
+        dest_iso: "{{ rebuilt_unattend_iso_path }}"
+        add_files:
+          - src_file: "{{ src_iso_file_dir }}/grub.cfg"
+            dest_file: "/boot/grub/grub.cfg"
+          - src_file: "{{ new_unattend_install_conf }}"
+            dest_file: "/preseed/ubuntu.seed"
+      when: 
+        - result_extract_file is defined
+        - result_extract_file.failed is defined
+        - result_extract_file.failed
+  when: unattend_install_conf is match('Ubuntu/Desktop/Ubiquity')
+
+- name: "Rebuild ISO for Ubuntu live server"
+  block:
+    - name: Extract specific files inside ISO
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - 'boot/grub/grub.cfg'
+          - 'md5sum.txt'
+
+    - name: "Add autoinstall to UEFI boot kernel command for Ubuntu live server"
+      ansible.builtin.replace:
+        path: "{{ unattend_iso_cache }}/grub.cfg"
+        regexp: '(.*vmlinuz)(.*)'
+        replace: "\\1 autoinstall \\2"
+      when: unattend_install_conf is match('Ubuntu/Server') or unattend_install_conf is match('Ubuntu/Desktop/Subiquity')
```

Comment:
I think this line of when: can be removed as it has same condition with the block when.

Reply:
Thanks. Done